source/extensions/common/wasm/wasm.h

```diff
@@ -141,6 +141,14 @@ class Wasm : public Logger::Loggable<Logger::Id::wasm>, public std::enable_share
     return true;
   }
 
+  void addAfterVmCallAction(std::function<void()> f) { after_vm_call_actions_.push_back(f); }
+  void doAfterVmCallActions() {
+    while (!after_vm_call_actions_.empty()) {
+      after_vm_call_actions_.back()();
+      after_vm_call_actions_.pop_back();
+    }
```

Comment:
This gets recursive and the deferred functions are called indefinitely before this code has a chance to remove them from the queue. I'm not sure why, to be honest.

One way to work around this is to make the call only after removing the function from the queue, i.e.
```
auto f = after_vm_call_actions_.back();
after_vm_call_actions_.pop_back();
f();
```
but that probably only works if the is only one deferred call?

Reply:
Done.

It should be fine  in the loop so long as the function to be called is removed first as you suggest.